src/impl-qti.c

```diff
@@ -203,20 +208,30 @@ static gboolean
 am_client_connect(
         AmClient* am)
 {
+    am->interfaceName = QCRIL_AUDIO_1_0;
+    am->interfaceCallbackName = QCRIL_AUDIO_CALLBACK_1_0;
     int status = 0;
+    am->fqname = g_strconcat(am->interfaceName, "/", am->slot, NULL);
     am->remote = gbinder_servicemanager_get_service_sync(am->sm,
         am->fqname, &status); /* auto-released reference */
-
+    if (!am->remote) {
+        DBG("vendor.qti.qcril.am failed, trying vendor.qti.hardware.radio.am...");
+        am->interfaceName = QCRIL_AUDIO_HW_RADIO_1_0;
+        am->interfaceCallbackName = QCRIL_AUDIO_HW_RADIO_CALLBACK_1_0;
+        am->fqname = g_strconcat(am->interfaceName, "/", am->slot, NULL);
```

Comment:
you should call `g_free(am->fqname);` before overwriting it.

Reply:
Well I am not sure, those pointers are not allocated by me but rather pointing to const char arrays. How would that need a free call?

Reply:
Ah you mean g_strconcat does allocate memory?